lxd/storage_zfs.go

```diff
@@ -625,22 +625,31 @@ func (s *storageZfs) ImageCreate(fingerprint string) error {
 		return err
 	}
 
+	cleanup := func(err error) error {
+		if zerr := s.zfsDestroy(fs); zerr != nil {
+			err = fmt.Errorf("%s  During cleanup: %s", err, zerr)
+		}
+		if shared.PathExists(subvol) {
+			if oserr := os.Remove(subvol); oserr != nil {
```

Comment:
I don't think so?  I'm pretty sure the only way subvol would be non-empty is if zfsDestroy failed, in which case I think both Remove and RemoveAll would fail due to "resource busy".

Hold off on merging this for now, I'll follow-up with the submitter of #2194 to find out what exactly was left laying around after the failed image pull.